src/test/java/io/confluent/examples/streams/interactivequeries/WordCountInteractiveQueriesExampleTest.java

```diff
@@ -236,6 +239,145 @@ public void shouldDemonstrateInteractiveQueries() throws Exception {
     assertTrue(keyValueBean.getKey().startsWith("streams"));
     assertThat(keyValueBean.getValue(), equalTo(3L));
   }
+  
+  @Test
+  public void shouldStartRestApiOnAnyValidHost() throws Exception {
```

Comment:
If all this test is verifying is that all the right binding and metadata advertisement happens, I think it would be better to skip the actual IQ logic test (the exiting test already checks it). Similar to your other test that verifies an exception binding to an invalid host, we could just do the setup and do a quick check that everything is hooked up properly.

Then again, I'm not very familiar with this example. Is this suggestion valid?

Reply:
@vvcephei Apart from verifying binding and metadata advertisement, We are testing all those scenarios(IQ logic test) just to ensure that all rest endpoints works as expected with this host as well.
If anyhow, they seem irrelevant, we can surely make the suggested changes.